docs/source/cli/graph-construction/single-machine-gconstruct.rst

```diff
@@ -3,181 +3,325 @@
+
+  Example:
+
+  .. code:: json
+
+    "transform": {"name": "bucket_numerical",
+                  "range": [10, 50],
+                  "bucket_cnt": 2,
+                  "slide_window_size": 10},
+
+* **No-op vector truncation (experimental)** truncates feature vectors to the length requested. The ``name`` field can be empty (e.g., ``{name: }``), and an integer ``truncate_dim`` value will determine the length of the output vector. This can be useful when experimenting with input features that were trained using `Matryoshka Representation Learning <https://arxiv.org/abs/2205.13147>`_.
+
+  Example:
+
+  .. code:: json
+
+    "transform": {"name": ,
+                  "truncate_dim": 24},
 
 .. _output-format:
 
-Output
-..........
-Currently, the graph construction pipeline outputs two output formats: ``DistDGL`` and ``DGL``. If select ``DGL``, the output is a file, named `<graph_name>.dgl` under the folder specified by the **-\-output-dir** argument, where `<graph_name>` is the value of argument **-\-graph-name**. If select ``DistDGL``, the output is a JSON file, named `<graph_name>.json`, and a set of `part*` folders under the folder specified by the **-\-output-dir** argument, where the `*` is the number specified by the **-\-num-parts** argument.
+Outputs of the graph consturction command
+............................................
+The graph construction command outputs two output formats: ``DistDGL`` and ``DGL`` specified by the argument **-\-output-format**. 
 
-By Specifying the output_format as ``DGL``, the output will be an `DGLGraph <https://docs.dgl.ai/en/1.0.x/generated/dgl.save_graphs.html>`_. By Specifying the output_format as ``DistDGL``, the output will be a partitioned graph named `DistDGL graph <https://doc.dgl.ai/guide/distributed-preprocessing.html#partitioning-api>`_. It contains the partitioned graph, a JSON config describing the meta-information of the partitioned graph, the mappings for the edges and nodes after partition, and other files that contain related metadata information, e.g., the new construction configuration JSON file that records the details of feature transformation operations.
+If select ``DGL``, the output is an `DGLGraph <https://docs.dgl.ai/en/1.0.x/generated/dgl.save_graphs.html>`_ file, named ``<graph_name>.dgl`` under the folder specified by the **-\-output-dir** argument, where `<graph_name>` is the value of argument **-\-graph-name**.
```

Comment:
Yes, it will generate id mapping files, two label state JSONs and a data transformation JSON, which should be equal to new construction configuration.

Updated.